### PR TITLE
add core-js to datastore getting started dependencies

### PR DESCRIPTION
**Pull request by chrisbonifacio**
### Instructions

**If this PR should not be merged upon approval for any reason, please submit as a DRAFT**

Which product(s) are affected by this PR (if applicable)?
- [ ] amplify-cli
- [ ] amplify-ui
- [ ] amplify-studio
- [ ] amplify-hosting
- [ ] amplify-libraries

Which platform(s) are affected by this PR (if applicable)?
- [ ] JS
- [ ] Swift
- [ ] Android
- [ ] Flutter
- [ ] React Native

**Please add the product(s)/platform(s) affected to the PR title**

#### Checks

- [ ] Does this PR conform to [the styleguide](https://github.com/aws-amplify/docs/blob/main/STYLEGUIDE.md)?

- [ ] Does this PR include filetypes other than markdown or images? Please add or update unit tests accordingly.

- [ ] Are any files being deleted with this PR? If so, have the needed redirects been created?

- [ ] Are all links in MDX files using the MDX link syntax rather than HTML link syntax? <br /> 
      _ref: MDX: `[link](https://docs.amplify.aws/)` 
            HTML: `<a href="https://docs.amplify.aws/">link</a>`_

### When this PR is ready to merge, please check the box below
- [ ] Ready to merge

_By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license._

### CODE REVIEW

**Review by david-mcafee — APPROVED** [review]
Thanks for updating this, @chrisbonifacio!

**Review by israx — DISMISSED** [review]

**Review by calebpollman — COMMENTED** [review]
IIRC `core-js` is only needed for a specific version(s) of RN for Android to polyfill AsyncIterator support when using DataStore ([see this SO issue](https://stackoverflow.com/questions/75986313/typeerror-symbol-asynciterator-is-not-defined-when-connecting-to-aws-amplify)), which may no longer be an issue in the RN versions currently supported by Amplify. 

Instead of adding the polyfill extraneously, the solution here is to verify that it is actually required, and if not remove the import step [here](https://docs.amplify.aws/react-native/build-a-backend/more-features/datastore/set-up-datastore/#add-required-polyfills). 

If a polyfill is still needed for DataStore usage, the documentation should reflect which versions of RN the polyfill is required for. If a polyfill for AsyncIterator support is required, `core-js` itself is a huge dep (1.21 MB) and can potentially be replaced with [@azure/core-asynciterator-polyfill](https://www.npmjs.com/package/@azure/core-asynciterator-polyfill) which is **much** lighter weight at 6.51 kb.

**Comment by reesscot** [comment]
> IIRC `core-js` is only needed for a specific version(s) of RN for Android to polyfill AsyncIterator support when using DataStore ([see this SO issue](https://stackoverflow.com/questions/75986313/typeerror-symbol-asynciterator-is-not-defined-when-connecting-to-aws-amplify)), which may no longer be an issue in the RN versions currently supported by Amplify.
> 
> Instead of adding the polyfill extraneously, the solution here is to verify that it is actually required, and if not remove the import step [here](https://docs.amplify.aws/react-native/build-a-backend/more-features/datastore/set-up-datastore/#add-required-polyfills).
> 
> If a polyfill is still needed for DataStore usage, the documentation should reflect which versions of RN the polyfill is required for. If a polyfill for AsyncIterator support is required, `core-js` itself is a huge dep (1.21 MB) and can potentially be replaced with [@azure/core-asynciterator-polyfill](https://www.npmjs.com/package/@azure/core-asynciterator-polyfill) which is **much** lighter weight at 6.51 kb.

+1, I think we should be more specific about when customers actually need this dependency rather than making all customers consume it.

**Comment by david-mcafee** [comment]
> IIRC `core-js` is only needed for a specific version(s) of RN for Android to polyfill AsyncIterator support when using DataStore ([see this SO issue](https://stackoverflow.com/questions/75986313/typeerror-symbol-asynciterator-is-not-defined-when-connecting-to-aws-amplify)), which may no longer be an issue in the RN versions currently supported by Amplify.
> 
> Instead of adding the polyfill extraneously, the solution here is to verify that it is actually required, and if not remove the import step [here](https://docs.amplify.aws/react-native/build-a-backend/more-features/datastore/set-up-datastore/#add-required-polyfills).
> 
> If a polyfill is still needed for DataStore usage, the documentation should reflect which versions of RN the polyfill is required for. If a polyfill for AsyncIterator support is required, `core-js` itself is a huge dep (1.21 MB) and can potentially be replaced with [@azure/core-asynciterator-polyfill](https://www.npmjs.com/package/@azure/core-asynciterator-polyfill) which is **much** lighter weight at 6.51 kb.

I tested an Expo RN app (using latest for all) with iOS, and it turns out the async iterator polyfill _is_ always required for DataStore. What are your thoughts on merging this PR to mitigate the issue (i.e. if customers follow the DataStore "Getting Started" guide as it is today, they will run into this), and exploring the alternative you linked above as a fast-follow?

File: src/fragments/lib-v1/datastore/react-native/getting-started/polyfills.mdx [hunk 1]
```diff
@@ -1,7 +1,7 @@
 ### Add Required Polyfills
 
-React Native is missing some JavaScript standard library functionality that DataStore depends on. Add the following to the entry point of your app to polyfill these missing functionalities:
+React Native or Expo apps using `react-native` 0.70 and above have Hermes enabled as the default engine and are missing some JavaScript standard library functionality that DataStore depends on. Add the following to the entry point of your app to polyfill these missing functionalities:
```
Comment (calebpollman) [inline src/fragments/lib-v1/datastore/react-native/getting-started/polyfills.mdx:3]: IMO we should avoid generic language regarding missing functionality:
```suggestion
React Native and Expo apps using `react-native` 0.70+ enable the Hermes engine by default, which does not currently support `asyncIterator` out of the box. DataStore requires adding the following to the entry point of your app to polyfill `asyncIterator` support:
```

File: src/fragments/lib/datastore/react-native/getting-started/polyfills.mdx [hunk 2]
```diff
@@ -1,7 +1,7 @@
 ### Add Required Polyfills
 
-React Native is missing some JavaScript standard library functionality that DataStore depends on. Add the following to the entry point of your app to polyfill these missing functionalities:
+React Native or Expo apps using `react-native` 0.70 and above have Hermes enabled as the default engine and are missing some JavaScript standard library functionality that DataStore depends on. Add the following to the entry point of your app to polyfill these missing functionalities:
```
Comment (calebpollman) [inline src/fragments/lib/datastore/react-native/getting-started/polyfills.mdx:3]: See https://github.com/aws-amplify/docs/pull/6774/files#r1453804445

**Review by calebpollman — APPROVED** [review]
🚢

**Pull request merged**